src/mc-fle2-tag-and-encrypted-metadata-block-private.h

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2024-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MC_FLE2_TAG_AND_ENCRYPTED_METADATA_BLOCK_H
+#define MC_FLE2_TAG_AND_ENCRYPTED_METADATA_BLOCK_H
+
+#include "mc-reader-private.h"
+#include "mc-writer-private.h"
+#include "mongocrypt-private.h"
+
+typedef struct {
+    _mongocrypt_buffer_t encryptedCount;
+    _mongocrypt_buffer_t tag;
+    _mongocrypt_buffer_t encryptedZeros;
+} mc_FLE2TagAndEncryptedMetadataBlock_t;
+
+#define kFieldLen 32U
+
+void mc_FLE2TagAndEncryptedMetadataBlock_init(mc_FLE2TagAndEncryptedMetadataBlock_t *metadata);
+
+void mc_FLE2TagAndEncryptedMetadataBlock_cleanup(mc_FLE2TagAndEncryptedMetadataBlock_t *metadata);
+
+bool mc_FLE2TagAndEncryptedMetadataBlock_parse(mc_FLE2TagAndEncryptedMetadataBlock_t *metadata,
+                                               mc_reader_t *reader,
```

Comment:
My inclination is, even though it's less efficient, the payload parse should read into a buffer and pass a const buffer into this parse.  The extra copy is small (96) bytes compared to the overall crypto op and it avoids spilling a random reader out.

Same on the serialize end...

@kevinAlbs wdyt?

Reply:
I went ahead and pushed these changes but put them in a single [commit](https://github.com/mongodb/libmongocrypt/pull/899/commits/b0e0fb2d11dea522bf43540bf6448c767b29b606). If we decide to not go ahead with this then it will be an easy revert.

Reply:
No strong opinion. I expect `mc_FLE2TagAndEncryptedMetadataBlock_parse` may only be useful for the server. If a buffer seems preferable, no objection from me.